specification/consumption/resource-manager/Microsoft.Consumption/preview/2019-04-01-preview/consumption.json

```diff
@@ -0,0 +1,3146 @@
+          "type": "string",
+          "readOnly": true
+        },
+        "reservationId": {
+          "description": "ARM resource id of the reservation. Only applies to records relevant to reservations.",
+          "type": "string",
+          "readOnly": true
+        },
+        "reservationName": {
+          "description": "User provided display name of the reservation. Last known name for a particular day is populated in the daily data. Only applies to records relevant to reservations.",
+          "type": "string",
+          "readOnly": true
+        },
+        "productOrderId": {
+          "description": "Product Order Id. For reservations this is the Reservation Order ID.",
+          "type": "string",
+          "readOnly": true
+        },
+        "productOrderName": {
+          "description": "Product Order Name. For reservations this is the SKU that was purchased.",
+          "type": "string",
+          "readOnly": true
+        },
+        "offerId": {
+          "description": "Offer Id. Ex: MS-AZR-0017P, MS-AZR-0148P.",
+          "type": "string",
+          "readOnly": true
+        },
+        "isAzureCreditEligible": {
+          "description": "Is Azure Credit Eligible.",
+          "type": "boolean",
+          "readOnly": true
+        },
+        "term": {
+          "description": "Term (in months). 1 month for monthly recurring purchase. 12 months for a 1 year reservation. 36 months for a 3 year reservation.",
+          "type": "string",
+          "readOnly": true
+        },
+        "publisherName": {
+          "description": "Publisher Name.",
+          "type": "string",
+          "readOnly": true
+        },
+        "publisherType": {
+          "description": "Publisher Type.",
+          "type": "string",
+          "readOnly": true
+        },
+        "planName": {
+          "description": "Plan Name.",
+          "type": "string",
+          "readOnly": true
+        },
+        "chargeType": {
+          "description": "Indicates a charge represents credits, usage, a Marketplace purchase, a reservation fee, or a refund.",
+          "type": "string",
+          "readOnly": true
+        },
+        "frequency": {
+          "description": "Indicates how frequently this charge will occur. OneTime for purchases which only happen once, Monthly for fees which recur every month, and UsageBased for charges based on how much a service is used.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "UsageDetailsDownloadResponse": {
+      "description": "Download response of Usage Details.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/DownloadUrl",
+          "title": "Download url properties."
+        }
+      }
+    },
+    "DownloadUrl": {
```

Comment:
This will be a breaking change to us since Ibizza is already using this (internally) and they are consuming the downloadUrl. We can change it may be in a later version with backward compatible. But for this version we would like to keep it like downloadUrl. Also this is the same model we are keeping in our 2018-11-01-preview pricesheets downlaod as well.
- https://github.com/Azure/azure-rest-api-specs/blob/0972d9b7ba5b90b1c730ae744bcc655414d825bb/specification/consumption/resource-manager/Microsoft.Consumption/preview/2018-11-01-preview/consumption.json

Reply:
Let's please keep it as-is. We're actively working on a new API to replace this and can align with ARM async download standards as part of that. That said, is there a defined standard? I've seen different patterns from different RPs. I'd hate to align to one nonstandard format only to be asked to align to another a few months later.